Atomization of the components

balintcristian opened the pull request:
De-structured components to atomic levels and moved them in corresponding directories.
Import arranged for readability.

balintcristian committed:
Atomization of the components

+import arrangement

balintcristian committed:
Added json for nav links.

In the Navbar component there's another code option at line one, which is to import from static directory which is inside the source directory

balintcristian committed:
Revert "Added json for nav links."
It should remove a feature from being present on the wrong branch
This reverts commit 169b9f2e4c2d69d60a33ae8b7fa41bc959c35d53.

robibalint commented on src/components/atoms/FooterIcon/FooterIcon.tsx:
```diff
@@ -0,0 +1,12 @@
+import Image from "next/image";
```
Let's make this a general Icon, not just for Footer

robibalint commented on src/components/molecules/FooterIconLink/FooterIconLink.tsx:
```diff
@@ -0,0 +1,19 @@
+import FooterIcon from "@/components/atoms/FooterIcon/FooterIcon";
+
```
Same. Let's make this IconLink

balintcristian replied:
On it, on the icon changes

robibalint commented on src/components/molecules/GalleryCard/GalleryCard.tsx:
```diff
@@ -0,0 +1,23 @@
+      width={0}
+      height={0}
+      sizes="100vw"
+      style={{ width: "100%", height: "100%" }}
```
Don't use inline styling. Let's stick to scss please. Why did you set `width` & `height` to 0 above?

balintcristian replied:
Cause i don't have de widths and heights and next js requires them for their image component. So i thought i'd be better to do it that way for now, until i find a better solution. Thanks for reminding me.

robibalint replied:
Then maybe stick to a plain `<img>` tag?

balintcristian replied:
Your call, it's a fair point after all. My point was that next js recommends those for better loading times. I can get the the inline style out though and make it a simple image i guess.

robibalint commented:
Also, shouldn't each of these components have a related scss file?

balintcristian committed:
Made footer icon into a general icon, component and footericonlink into a general icon link

balintcristian commented:
Working on the scss files as we speak

balintcristian committed:
Various changes

Created scss modules for each file.
Changed nextjs Images to regular image tags.
Added import for each scss module even if not used yet.
Added option to add descriptions to images.

balintcristian commented:
Renamed GalleryCard to GalleryImage, the whole code was written as GalleryImage and removed the inline stylings. Though, i could go back to gallery card.
